### PR TITLE
[SPARK-21439][PySpark] Support for ABCMeta in PySpark

### PR DESCRIPTION
**Pull request by maver1ck**
https://github.com/cloudpipe/cloudpickle/pull/104/files

## What changes were proposed in this pull request?

Possibility to use ABCMeta with Spark.

## How was this patch tested?

Manual tests

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@maver1ck, would you mind resolving the conflicts? I will try to review this.

**Comment by HyukjinKwon** [comment]
gentle ping @maver1ck.

File: python/pyspark/cloudpickle.py [hunk 1]
```diff
@@ -667,6 +667,13 @@ def save_file(self, obj):
     else:
         dispatch[file] = save_file
 
+    # WeakSet was added in 2.7.
```
Comment (HyukjinKwon) [inline python/pyspark/cloudpickle.py]: BTW, I'd add a simple e2e test.

**Comment by maver1ck** [comment]
Ping received.
I'll try to add tests and resolve conflict

**Comment by HyukjinKwon** [comment]
This should be good to go as soon as updated.

**Comment by maver1ck** [comment]
I realized that changes here was added also in SPARK-21070.
https://github.com/apache/spark/commit/751f513367ae776c6d6815e1ce138078924872eb

So we can close this PR.

**Comment by HyukjinKwon** [comment]
Let's close this then :).

**Pull request closed without merging**